### PR TITLE
protected adoptions api

### LINKED ISSUE #159: Setup authorization
depends on https://github.com/animavita/animavita/pull/135

### PR DESCRIPTION
**Pull request by imbhaskarn**
closes #159

### CODE REVIEW

File: apps/backend/src/adoptions/adoptions.controller.ts [hunk 1]
```diff
@@ -15,29 +16,34 @@ import {
 } from '@nestjs/common';
 import { JoiValidationPipe } from '../pipes/joi-validation-pipe';
 import { AdoptionsService } from './adoptions.service';
+import { AccessTokenGuard } from 'src/guards/accessToken.guard';
 
 @Controller('api/v1/adoptions')
 export class AdoptionsController {
   constructor(private readonly adoptionsService: AdoptionsService) {}
-
+
```
Comment (belgamo) [inline apps/backend/src/adoptions/adoptions.controller.ts]: You can remove this space 💅

File: apps/backend/src/adoptions/adoptions.controller.ts [hunk 2]
```diff
@@ -15,29 +16,34 @@ import {
 } from '@nestjs/common';
 import { JoiValidationPipe } from '../pipes/joi-validation-pipe';
 import { AdoptionsService } from './adoptions.service';
+import { AccessTokenGuard } from 'src/guards/accessToken.guard';
```
Comment (belgamo) [inline apps/backend/src/adoptions/adoptions.controller.ts]:
```suggestion
import { AccessTokenGuard } from '../guards/accessToken.guard';
```

**Review by belgamo — CHANGES_REQUESTED** [review]
Hey, thanks a lot for this PR! I left some comments and you also need to fix the e2e specs so we can merge the PR 😉

File: apps/backend/src/adoptions/adoptions.controller.ts (hunk 2 above)
Comment (imbhaskarn) [reply apps/backend/src/adoptions/adoptions.controller.ts]: Hey @belgamo Thanks for suggentions. I'm getting typescript error for this import. Should I ignore this ? 
Cannot find module '../src/guards/accessToken.guard' or its corresponding type

Comment (belgamo) [reply apps/backend/src/adoptions/adoptions.controller.ts]: Please try `../guards/accessToken.guard` without the `src`

Comment (imbhaskarn) [reply apps/backend/src/adoptions/adoptions.controller.ts]: @belgamo  I fixed the import path. This import was also failing e2e specs.

**Review by belgamo — COMMENTED** [review]
Thanks for fixing the import error. You still need to fix the [e2e tests](https://github.com/animavita/animavita/actions/runs/4514064827/jobs/7959700440?pr=168) so we can proceed to merge.

**Comment by belgamo** [comment]
Closing due inactivity

**Pull request closed without merging**